packages/core/src/crawlers/error_snapshotter.ts

```diff
@@ -0,0 +1,135 @@
+                context as unknown as
+                | PlaywrightCrawlingContext
+                | PuppeteerCrawlingContext,
+                filename,
+            );
+
+            if (capturedFiles) {
+                screenshotFilename = capturedFiles.screenshotFilename;
+                htmlFileName = capturedFiles.htmlFileName;
+            }
+
+            // If the snapshot for browsers failed to capture the HTML, try to capture it from the page content
+            if (!htmlFileName) {
+                const html = await page?.content() || undefined;
+                htmlFileName = html ? await this.saveHTMLSnapshot(html, keyValueStore, filename) : undefined;
+            }
+        } else if (typeof body === 'string') { // for non-browser contexts
+            htmlFileName = await this.saveHTMLSnapshot(body, keyValueStore, filename);
+        }
+
+        if (APIFY_IS_AT_HOME) {
+            const platformPath = `${KEY_VALUE_PLATFORM_PATH}/${keyValueStore.id}/records`;
+            return {
+                screenshotFileUrl: screenshotFilename ? `${platformPath}/${screenshotFilename}` : undefined,
+                htmlFileUrl: htmlFileName ? `${platformPath}/${htmlFileName}` : undefined,
+            };
+        }
+
+        const localPath = `${KEY_VALUE_STORE_LOCAL_PATH}/${keyValueStore.name || 'default'}`;
+        return {
+            screenshotFileUrl: screenshotFilename ? `${localPath}/${screenshotFilename}` : undefined,
+            htmlFileUrl: htmlFileName ? `${localPath}/${htmlFileName}` : undefined,
+        };
+    }
+
+    /**
+     * Capture a screenshot and HTML of the page (For Browser only), and return the filename with the extension.
+     */
+    async captureSnapShot(
+        context: PlaywrightCrawlingContext | PuppeteerCrawlingContext,
+        filename: string): Promise<{
+            screenshotFilename?: string;
+            htmlFileName?: string;
+        } | undefined> {
+        try {
+            await context.saveSnapshot({ key: filename });
+            return {
+                // The screenshot file extension is different for Apify and local environments
+                screenshotFilename: `${filename}${APIFY_IS_AT_HOME ? '.jpg' : '.jpeg'}`,
+                htmlFileName: `${filename}.html`,
+            };
+        } catch (e) {
+            return undefined;
+        }
+    }
+
+    /**
+     * Save the HTML snapshot of the page, and return the filename with the extension.
+     */
+    async saveHTMLSnapshot(html: string, keyValueStore: KeyValueStore, filename: string): Promise<string | undefined> {
+        try {
+            await keyValueStore.setValue(filename, html, { contentType: 'text/html' });
+            return `${filename}.html`;
+        } catch (e) {
+            return undefined;
+        }
+    }
+
+    /**
+     * Remove non-word characters from the start and end of a string.
+     */
+    sanitizeString(str: string): string {
+        return str.replace(/^\W+|\W+$/g, '');
```

Comment:
how is this different from `str.trim()`? if its the same, there is no point in having such helper. also it should be probably private

Reply:
This is different
`('-apify_').replace(/^\W+|\W+$/g, '')` => `'apify_'`
`('-apify_').trim()` => `'-apify_'`

It is because Apify forbids records to start with `-` but I tested it now and `await Actor.setValue('-apify_', 12)` so perhaps this is not needed at all. I think that start/end restriction was either removed or only applies to KV Store name itself.

See the current error on what must be sanizitzed 
```
The "key" argument "$%#$^" must be at most 256 characters long and only contain the following characters: a-zA-Z0-9!-_.'()
```